src/SplitGraph/energy.jl

```diff
@@ -208,17 +213,24 @@ function doMCMC(g::Graph, conf::Array{Bool,1}, M::Int64; T=1, γ=1)
 end
 
 """
+	mcmcstep!(conf, g, F, T, γ; mask=Set(), constraint_cost=γ)
+
+Perform an mcmc step by removing a node from `conf` at random.
+If `mask` is specified masked nodes will be removed at cost `constraint_cost`,
+per default this is set to the cost of a reassortment event.
 """
-function mcmcstep!(conf, g, F, T, γ)
-	i = rand(1:length(conf))
+function mcmcstep!(conf, g, F, T, γ; mask=Set(), constraint_cost=γ)
+	s = Set(1:length(conf))
+	i = rand(s)
 	conf[i] = !conf[i]
 	Enew = compute_energy(conf, g)
-	Fnew = Enew + γ*(length(conf) - sum(conf))
+	c = sum([Int(i ∈ mask && !conf[i]) for i in 1:length(conf)])*constraint_cost
+	Fnew = Enew + γ*(length(conf) - sum(conf)) + c
 	if Fnew < F || exp(-(Fnew-F)/T) > rand()
 		return Enew, Fnew
 	else
 		conf[i] = !conf[i]
-		return (round(Int64, F-γ*(length(conf) - sum(conf))), F)
+		return (round(Int64, F-γ*(length(conf) - sum(conf))-c), F)
```

Comment:
Why do we round to `Int` here? I know I had this before but I can't see the logic of it now :)